src/query.c

```diff
@@ -1236,6 +1236,14 @@ static IndexIterator *query_EvalSingleTagNode(QueryEvalCtx *q, TagIndex *idx, Qu
     tag_strtolower(n->tn.str, &n->tn.len, fs->tagOpts.tagFlags & TagField_CaseSensitive);
   }
 
+  // Support for EMPTY TAG values (searched via the "__empty" token).
+  if (n->tn.str && strcmp(n->tn.str, "__empty") == 0 && FieldSpec_IndexesEmpty(fs)) {
```

Comment:
Right. No we currently don't allow it (syntax error) and it's not planned for this PR. I think we went for the `__empty` keyword in order to avoid ambiguity in the query, but we can reconsider.